src/pages/blog.tsx

```diff
@@ -2,29 +2,42 @@ import { graphql } from 'gatsby';
 import React from 'react';
 import BlogCard from '../components/BlogCard';
 import Layout from '../components/Layout';
-import { BlogPostsList, BlogCategoriesList } from '../types';
+import SideNavBar from '../components/SideNavBar';
+import { BlogPostsList, BlogCategoriesList, SideNavBarItem } from '../types';
 
 type Props = {
   data: BlogPostsList & BlogCategoriesList;
 };
 
-const Blog = ({ data: { posts } }: Props): JSX.Element => (
+export const getNavigationData = (
```

Comment:
This transforms the blog categories into data that the `SideNavBar` Component can use. It could also reside in a different file, like `utils` but I have no strong opinions here.